src/Bicep.Core/Navigation/SyntaxBaseExtensions.cs

```diff
@@ -78,6 +80,42 @@ public static string ToText(this SyntaxBase syntax, string indent="")
             document.Layout(sb, indent, System.Environment.NewLine);
             return sb.ToString();
         }
+
+        /// <summary>
+        /// Generate a string that represents this Syntax element optionally preserving formatting.
+        /// </summary>
+        public static string ToTextPreserveFormatting(this SyntaxBase syntax)
```

Comment:
>ToTextPreserveFormatting [](http://example.com/codeflow?start=29&length=24)

In \src\Bicep.Core.IntegrationTests\ParserTests.cs, we have `FilesShouldRoundTripSuccessfully()` and `Oneliners_ShouldRoundTripSuccessfully()`. Can you update them to also assert that the result from `ToTextPreserveFormatting()` matches what the tests get from their own print visitor?

Reply:
I was trying to remove the last usage of PrintVisitor and stumbled on this test code in PrintProgram_AnyProgram_ShouldRoundTrip:

```
			printVisitor.Visit(program);
            string programText = buffer.ToString();

            buffer.Clear();
            printVisitor.Visit(program);
            string formattedProgramText = buffer.ToString();

            formattedProgramText.Should().Be(programText);
```

Am I missing something or does it compare two equivalent outputs of printVisitor?

Reply:
I think it should've been:

```

            printVisitor.Visit(formattedProgram);
            string formattedProgramText = buffer.ToString();
```

Reply:
Ok, I won't try to fix it in this PR, but FYI it starts failing if you try to format the formattedProgram

Reply:
This might be a good question for @shenglol who did the formatter.

Reply:
Ahh good catch. That's definitely a bug. I've created https://github.com/Azure/bicep/issues/4605 to track it and will try to fix it later.

Reply:
@shengol I ended up fixing the test in this PR (https://github.com/Azure/bicep/pull/4570/files/90f961f9042ae2e8ed9b46712c9172d33d0c6cd8#diff-026bc8f7b01e7149a98805711f52ea0f2df9cc6f0664ad061d6f982358cbe5d4) to get rid of test PrintVisitor.

Reply:
Awesome! Thanks for fixing it. I've assigned https://github.com/Azure/bicep/issues/4605 to you and linked it.